examples/permissioned-token/assets/clawback.py

```diff
@@ -0,0 +1,116 @@
+        Gtxn[1].close_remainder_to() == Global.zero_address(),
+        Gtxn[2].close_remainder_to() == Global.zero_address(),
+        Gtxn[0].asset_close_to() == Global.zero_address(),
+        Gtxn[1].asset_close_to() == Global.zero_address(),
+        Gtxn[2].asset_close_to() == Global.zero_address()
+    )
+
+    # verify first transaction
+    # check level smart contract call - signed by asset sender
+    first_transaction_checks = And(
+        # check app_id passed through params
+        Gtxn[0].application_id() == Int(CONTROLLER_APP_ID),
+        Gtxn[0].sender() == Gtxn[2].sender(),
+        Gtxn[0].sender() == Gtxn[1].asset_sender()
+    )
+
+    # verify second transaction
+    # tx 1 - clawback transactions that moves the frozen asset from sender to receiver - signed by clawback
+    # verify the account sent in the accounts array is
+    # actually the receiver of the asset in asset xfer
+    second_transaction_checks = (
+        # check asset_id passed through params
+        Gtxn[1].xfer_asset() == Int(TOKEN_ID)
+    )
+
+    # verify third transaction
+    # tx 2 - payment transaction from sender to clawback to pay for the fee of the clawback
+    third_transaction_checks = And(
+        Gtxn[1].sender() == Gtxn[2].receiver(),
+        # verify the fee amount is good
+        Gtxn[2].amount() >= Gtxn[1].fee()
+    )
+
+    # transfer of token between non-reserve accounts
+    # this must be passed with rule checks
+    token_transfer = And(
+        group_tx_checks,
+        common_fields,
+        first_transaction_checks,
+        second_transaction_checks,
+        third_transaction_checks
+    )
+
+    # issue new token (asset transfer tx from asset reserve to receiver)
+    # since issuer creates the rules, this can be a simple asset transfer transaction (without rule checks)
+    issuance_tx = And(
+        Gtxn[0].type_enum() == TxnType.ApplicationCall,
+        Gtxn[0].application_id() == Int(CONTROLLER_APP_ID),
+        Gtxn[0].sender() == Gtxn[1].asset_sender(),
+        Gtxn[0].rekey_to() == Global.zero_address(),
+
+        Gtxn[1].type_enum() == TxnType.AssetTransfer,
+        Gtxn[1].xfer_asset() == Int(TOKEN_ID), # verify token index
+        Gtxn[1].rekey_to() == Global.zero_address(),
+    )
+
+    update_reserve = And(
+        # in 1st tx we move all funds to new asset reserve
+        Gtxn[0].type_enum() == TxnType.AssetTransfer,
+        Gtxn[0].xfer_asset() == Int(TOKEN_ID), # verify token index for asset_transfer
+
+        # imp: check asset receiver first tx (move funds) is the new reserve
+        Gtxn[0].asset_receiver() == Gtxn[1].config_asset_reserve(),
+
+        Gtxn[1].type_enum() == TxnType.AssetConfig,
+        Gtxn[1].config_asset() == Int(TOKEN_ID), # verify token index for asset_config
+
+        # ensure no rekeying
+        Gtxn[0].rekey_to() == Global.zero_address(),
+        Gtxn[1].rekey_to() == Global.zero_address()
+    )
+
```

Comment:
we need to add rule for `force_transfer` from the [spec](https://paper.dropbox.com/doc/Algob-Security-Token-Template--BKDPSFg1YRt3TQQjWHyScz4QAg-FR2LXhVg3edevYPBQZw6F#:uid=481217053399840855734627&h2=Issuance-Functions).  (I've renamed the `clawback` function to `force_transfer` to not confuse with the clawback asset attribute).

Reply:
yes, done.